newsService/src/test/java/com/kovan/NewsServiceTest.java

```diff
@@ -47,7 +50,7 @@ void testGetTopHeadlines_Success() throws Exception {
         when(restTemplate.getForObject(anyString(), eq(String.class))).thenReturn(response);
         when(objectMapper.readValue(anyString(), eq(NewsDto.class))).thenReturn(newsDto);
 
-        when(newsRepository.findByPublishedAt("2023-10-10")).thenReturn(Optional.empty());
+        when(newsRepository.findByPublishedAt("2023-10-10")).thenReturn(newsEntity);
```

Comment:
find any other way to return entity. avoid using mock for entities, pojo's

Reply:
Changed from returning newsEntity to null.